R/track.R

```diff
@@ -0,0 +1,403 @@
+#' Compute track statistics
+#'
+#' @param population, single cell data
+#' @param grouping_variable, column name storing the track label
+#' @return track
+#' @importFrom magrittr %>%
+#' @examples 
+#'  data <- tibble::data_frame(
+#'    Metadata_timePoint = c(1:5),
+#'    Location_Center_X = c(1, 2, 3, 4, 5),
+#'    Location_Center_Y = c(1, 1, 1, 1, 1),
+#'    TrackObjects_Label = c(rep(1, 5))
+#'  )
+#'  data <- dplyr::group_by_(data,'TrackObjects_Label')
+#'  tracks <- track(data,'TrackObjects_Label')
+#' @export
+track <- function(population, grouping_variable) {
+  # process `population`, which is the data you get from CellProfiler
+  tracks <- displace(population, grouping_variable)
+  
+  features <- list(
+    angle(tracks),
+    chemotacticIndex(tracks),
+    directionality(tracks),
+    distance(tracks),
+    directionalPersistence(tracks),
+    forwardMigrationIndex(tracks),
+    lifeTime(tracks),
+    meanSquaredDisplacement(tracks,tau = 2),
+    sectorAnalysis(tracks),
+    speed(tracks)) 
+  
+  return(Reduce(function(...) merge(..., all = TRUE, by = grouping_variable), features))
+}
+
+#' Add spatial displacement per frame for each track object
+#'
+#' @param population, data frame storing single cell data 
+#' @param group_variable, column name storing the track label
+#' @return displacement   
+#' @examples 
+#'  data <- tibble::data_frame(
+#'    Metadata_timePoint = c(1:5),
+#'    Location_Center_X = c(1, 2, 3, 4, 5),
+#'    Location_Center_Y = c(1, 1, 1, 1, 1),
+#'    TrackObjects_Label = c(rep(1, 5))
+#'  )
+#'  tracks <- cytominer::displace(data,'TrackObjects_Label')
+#' @importFrom magrittr %>%
+#' @export
+displace <- function(population, group_variable) {
+  dplyr::right_join(
+    population %>%
+      dplyr::select_(.dots = c(group_variable, 'Location_Center_X', 'Location_Center_Y','Metadata_timePoint')) %>%
+      dplyr::mutate(Metadata_timePoint2 =  (Metadata_timePoint - 1) ) %>% 
```

Comment:
@shntnu We need to make a decision about whether we should or should not use CellProfiler feature names. We should probably force the user to pass the appropriate arguments rather than collect them from some parent data structure.